docker/ci/build.ps1

```diff
@@ -19,27 +19,46 @@ function Get-ContainerVersion
   (git describe) -replace '-.*', ''
 }
 
-# installs gems for build and test and grabs base images
-function Invoke-ContainerBuildSetup
+function Lint-Dockerfile($Path)
 {
-  Push-Location (Get-CurrentDirectory)
-  bundle install --path '.bundle/gems'
-  bundle exec puppet-docker update-base-images ubuntu:16.04
-  Pop-Location
+  hadolint --ignore DL3008 --ignore DL3018 --ignore DL4000 --ignore DL4001 $Path
 }
 
-function Build-Container($Name, $Repository = '127.0.0.1', $Version = (Get-ContainerVersion))
+function Build-Container(
+  $Name,
+  $Namespace = 'puppet',
+  $Version = (Get-ContainerVersion),
+  $Vcs_ref = $(git rev-parse HEAD))
 {
   Push-Location (Join-Path (Get-CurrentDirectory) '..')
-  bundle exec puppet-docker local-lint $Name
-  bundle exec puppet-docker build $Name --no-cache --repository $Repository --version $Version --no-latest --build-arg namespace=$Repository
+
+  $build_date = (Get-Date).ToUniversalTime().ToString('o')
+  $docker_args = @(
+    '--pull',
```

Comment:
we don't always want `--pull`, we need to build the puppetserver container based on the puppetserver-standalone container we just built.

Reply:
Did you mean to comment on the Makefile change where I added the second `--pull`?  Or do we really not want to do any pulling here in the powershell?

Reply:
Well, we want pulling in the powershell, but it should be conditional so we aren't trying to pull when building the puppetserver container but are for the puppetserver-standalone.

Reply:
How was that handled before this PR?  I don't remember seeing anything around conditional building.

Reply:
We weren't using `--pull` prior to this, we were explicitly pulling the base image for the puppetserver-standalone container.

Reply:
I think I fixed this in: https://github.com/puppetlabs/puppetserver/pull/1934/commits/f7e0911306bc447aa18be537c51ea788aca8fc01